### PR TITLE
Document `stages` for new hooks

### LINKED ISSUE #283: Undocumented `stages` property under "Creating new hooks"
The [Creating new hooks](https://pre-commit.com/#new-hooks) section in the docs does not mention the `stages` property that is used by some of the pre-commit hooks such as [trailing-whitespace](https://github.com/pre-commit/pre-commit-hooks/blob/5df1a4bf6f04a1ed3a643167b38d502575e29aef/.pre-commit-hooks.yaml#L194).

### PR DESCRIPTION
**Pull request by boidolr**
Fixes #283 by adding the description for `stages` from adding plugins to creating hooks.

### CODE REVIEW

File: index.mako [hunk 1]
```diff
@@ -452,7 +452,7 @@ file that tells pre-commit:
         single process instead of in parallel. _new in 1.13.0_.
 =r=
     =c= [`description`](_#hooks-description)
-    =c= (optional: default `''`) description of the hook.  used for metadata
+    =c= (optional: default `''`) description of the hook. Used for metadata
```
Comment (asottile) [inline index.mako:455]: please leave this as it was

File: index.mako [hunk 2]
```diff
@@ -465,6 +465,12 @@ file that tells pre-commit:
 =r=
     =c= [`args`](_#hooks-args)
     =c= (optional: default `[]`) list of additional parameters to pass to the hook.
+=r=
+    =c= [`stages`](_#hooks-stages)
+    =c= (optional) confines the hook to the `commit`, `merge-commit`, `push`,
```
Comment (asottile) [inline index.mako:470]: this should specify the default (I'd copy that from `default_stages`)

File: index.mako [hunk 3]
```diff
@@ -467,8 +467,8 @@ file that tells pre-commit:
     =c= (optional: default `[]`) list of additional parameters to pass to the hook.
 =r=
     =c= [`stages`](_#hooks-stages)
-    =c= (optional) confines the hook to the `commit`, `merge-commit`, `push`,
-        `prepare-commit-msg`, `commit-msg`, `post-checkout`, `post-commit`, or
+    =c= (optional: default `[commit, push]`) confines the hook to the `commit`, `merge-commit`,
```
Comment (asottile) [inline index.mako:470]: no no, I mean copy the text from there, the default is all stages just like `default_stages`: https://pre-commit.com/#top_level-default_stages

Comment (boidolr) [reply index.mako:470]: Oh sure, sorry about that.

**Review by asottile — APPROVED** [review]

**Pull request merged**